frontend/app.js

```diff
@@ -18,94 +18,86 @@
  *
  */
 
-const path = require('path');
-const crypto = require('crypto');
-const bodyParser = require('body-parser');
-const express = require('express');
-const probe = require('kube-probe');
-const rhea = require('rhea');
-
-// AMQP
-
-const amqpHost = process.env.MESSAGING_SERVICE_HOST || 'localhost';
-const amqpPort = process.env.MESSAGING_SERVICE_PORT || 5672;
-const amqpUser = process.env.MESSAGING_SERVICE_USER || 'work-queue';
-const amqpPassword = process.env.MESSAGING_SERVICE_PASSWORD || 'work-queue';
-
-const id = 'frontend-nodejs-' + crypto.randomBytes(2).toString('hex');
-const container = rhea.create_container({id});
-
-let connection = null;
-
-const requestIds = [];
-const responses = {};
-const workers = {};
-
-let requestSequence = 0;
-
-function sendRequest (message) {
-    message.to = 'work-queue-requests';
-    connection.send(message);
-    console.log(`${id}: Sent request ${JSON.stringify(message)}`);
+const path = require('path')
+const bodyParser = require('body-parser')
+const express = require('express')
+const boom = require('@hapi/boom')
+const probe = require('kube-probe')
+const exphbs = require('express-handlebars')
+const log = require('./lib/log')
+const { getUsernameFromRequest } = require('./lib/utils')
+const { getKeycloakInstance, getSessionMiddleware, threescaleApiMiddleware } = require('./lib/middleware')
+
+const app = express()
+const keycloak = getKeycloakInstance()
+
+// Required when running behind a load balancer, e.g HAProxy
+app.set('trust proxy', true)
```

Comment:
Required for SSO redirect URIs (when using Keycloak) to be constructed correctly.